sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs

```diff
@@ -11,11 +11,11 @@
 namespace Azure.ResourceManager.NotificationHubs.Models
 {
     /// <summary> Parameters supplied to the Check Name Availability for Namespace and NotificationHubs. </summary>
-    public partial class CheckAvailabilityParameters : TrackedResourceData
+    public partial class NotificationHubNameAvailabilityContent : TrackedResourceData
```

Comment:
This model is strange. Usually we will have a name and resource type property in NameAvailabilityContent, why does this model contain sku and IsAvailiable? And it inherits from TrackedResourceData, which I think is not reasonable.

Reply:
Good catch, check the swagger I found this is not check the `name` but the resource, so this is why contains `sku` and other properties, and the inherit is caused by codegen's property reference match

Reply:
Remove the `Name` from the model name to `NotificationHubAvailabilityContent`

Reply:
Also renamed the corresponding methods